RUFAS/routines/field/crop/crop.py

```diff
@@ -136,3 +140,219 @@ def grow_crop(
+
+    def get_canopy_water_excess_capacity(self) -> float:
+        """
+        Returns the excess capacity of the canopy water storage.
+        This is the difference between the storage capacity and the current water stored in the canopy.
+
+        Returns
+        -------
+        float
+            The excess capacity in the canopy (can be negative if over capacity).
+        """
+        return self.data.water_canopy_storage_capacity - self.data.canopy_water
+
+    def calculate_canopy_excess_water(self, canopy_water_excess_capacity: float) -> float:
+        """
+        Calculates the excess water in the canopy based on the canopy storage capacity.
+
+        Parameters
+        ----------
+        canopy_water_excess_capacity : float
+            The excess capacity of the canopy.
+
+        Returns
+        -------
+        float
+            The excess water in the canopy (negative if there is excess).
+        """
+        return min(0.0, canopy_water_excess_capacity)
+
+    def adjust_canopy_water_for_excess(self, excess_water_in_canopy: float) -> None:
+        """
+        Adjusts the canopy water based on the calculated excess.
+
+        Parameters
+        ----------
+        excess_water_in_canopy : float
+            The excess water in the canopy to adjust (negative value).
+        """
+        if excess_water_in_canopy != 0.0:
+            self.data.canopy_water = self.data.water_canopy_storage_capacity
+
+    def store_water_in_canopy(self, canopy_water_excess_capacity: float, precipitation_reaching_soil: float) -> float:
+        """
+        Stores water in the canopy and returns the amount of precipitation that remains to reach the soil.
+
+        Parameters
+        ----------
+        canopy_water_excess_capacity : float
+            The excess capacity of the canopy.
+        precipitation_reaching_soil : float
+            Amount of precipitation available to be stored in the canopy (mm).
+
+        Returns
+        -------
+        float
+            The amount of precipitation left after storing in the canopy (mm).
+        """
+        water_taken_to_be_stored = min(precipitation_reaching_soil, max(0.0, canopy_water_excess_capacity))
+        self.data.canopy_water += water_taken_to_be_stored
+        return precipitation_reaching_soil - water_taken_to_be_stored
+
+    def evaporate_from_canopy(self, evapotranspirative_demand: float) -> float:
+        """
+        Evaporates water from the crop's canopy based on the evapotranspirative demand.
+
+        Parameters
+        ----------
+        evapotranspirative_demand : float
+            The evapotranspirative demand on the field on the current day (mm).
+
+        Returns
+        -------
+        float
+            The amount of water evaporated from the crop's canopy (mm).
+        """
+        amount_evaporated = self.water_dynamics.evaporate_from_canopy(evapotranspirative_demand)
+        return amount_evaporated
```

Comment:
```suggestion
    def evaporate_from_canopy(self, evapotranspirative_demand: float) -> float:
        """Wrapper for the canopy evaporation routine."""
        return self.water_dynamics.evaporate_from_canopy(evapotranspirative_demand)
```
I think it would help make the code more readable/succinct to not include a full docstring for this method.